airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -70,10 +70,16 @@ def __post_init__(self, parameters: Mapping[str, Any]) -> None:
                 f"If step is defined, cursor_granularity should be as well and vice-versa. "
                 f"Right now, step is `{self.step}` and cursor_granularity is `{self.cursor_granularity}`"
             )
-        if not isinstance(self.start_datetime, MinMaxDatetime):
-            self.start_datetime = MinMaxDatetime(self.start_datetime, parameters)
-        if self.end_datetime and not isinstance(self.end_datetime, MinMaxDatetime):
-            self.end_datetime = MinMaxDatetime(self.end_datetime, parameters)
+        self._start_datetime = (
+            MinMaxDatetime(self.start_datetime, parameters) if not isinstance(self.start_datetime, MinMaxDatetime) else self.start_datetime
```

Comment:
I don't know how exactly `MinMaxDatetime` works, but could we make it's constructor always return an object of `MinMaxTime` and accept `MinMaxtime` too so that we can always wrap start_datetime into it, cutting on one if statement? 

Sure, that's one more function call, but perhaps more readable?

Reply:
good idea. Added a factory method for `MinMaxDatetime` and simplified this call